### PR TITLE
Global Styles: Expose the provider and the reset hook

### PR DESCRIPTION
**Pull request by scruffian**
## What?
This exports the GlobalStylesProvider component and the useGlobalStylesReset hook, so that plugins can interact with Global Styles.

## Why?
This will be used in the create block theme plugin: https://github.com/WordPress/create-block-theme/pull/96

## How?
Simple - export these functions from the edit-site package.

## Testing Instructions
See https://github.com/WordPress/create-block-theme/pull/96

### CODE REVIEW

File: schemas/json/block.json
Comment (scruffian) [inline schemas/json/block.json]: This is an unrelated change.

**Review by jffng — COMMENTED** [review]
Should it be documented [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/global-styles/README.md
) if it's being exposed as a public API? 

I find this quite useful for the purposes of letting plugin developers interact with global styles, as demonstrated by https://github.com/WordPress/create-block-theme/pull/96/ !

**Comment by scruffian** [comment]
> Should it be documented [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/global-styles/README.md) if it's being exposed as a public API?

Good idea, done!

File: packages/edit-site/src/components/global-styles/README.md
Comment (jffng) [inline packages/edit-site/src/components/global-styles/README.md]: I think this hook is already documented above? :)

**Comment by scruffian** [comment]
Oops you're right, it's gone!

**Comment by pbking** [comment]
Would really love to see this land in 6.1 to enable tools like [Create Block Theme](https://github.com/WordPress/create-block-theme/pull/96) to progress.  Are there any reasons why this can't be brought in?

**Comment by scruffian** [comment]
> Would really love to see this land in 6.1 to enable tools like https://github.com/WordPress/create-block-theme/pull/96 to progress. Are there any reasons why this can't be brought in?

Just need an approval from someone!

**Review by ajlende — APPROVED** [review]
I don't see any problem with exposing this publicly—it seems pretty stable to me.

I would still like to hear from @oandregal or @jorgefilipecosta—they've both seen the major refactoring this component has gone through and may have some thoughts on the stability.

**Comment by oandregal** [comment]
:wave: My first instinct is that exporting only the two functions that are used by `create-block-theme` is a bit custom. I'd suggest looking at how _any_ plugin should interact with GS (if they should be allowed to?).

I know Jorge & Riad discussed options for extracting GS to its own package. Jorge has a wider view of the extensibility systems in place than I have. I'd like to here what his thoughts are before merging this.

**Review by jorgefilipecosta — COMMENTED** [review]
I think we are not ready to have a public global styles API yet. Right now, global styles are part of the edit site. As @oandregal referred, we may have a global styles package in the future, if we also include parts of global styles on edit-post.
I also imagine that if we are going to expose an API for global styles, we would not expose the provider and reset directly but instead expose a single hook that provides global styles value and a setter.

**Comment by jorgefilipecosta** [comment]
I imagine not having an API may be a blocker. I guess a possible intermediate step would be to expose a clearly experimental, internal, and unstable API so we can have some progress on https://github.com/WordPress/create-block-theme/pull/96.

**Comment by scruffian** [comment]
Thanks for the feedback @jorgefilipecosta, I marked them as unstable...

**Comment by ajlende** [comment]
The exports should be named `__unstable*` too

**Comment by scruffian** [comment]
Closing this old PR.

**Pull request closed without merging**
